docs/decisions/0009-course-catalog-info-changed-design.rst

```diff
@@ -0,0 +1,37 @@
+8. Event design of the COURSE_CATALOG_INFO_CHANGED signal
+=========================================================
+
+Status
+------
+
+Provisional
+
+Context
+-------
+
+The COURSE_CATALOG_INFO_CHANGED signal is used by Studio to convey that information relevant to the course catalog has changed. It was created with an eye towards eventually replacing the refresh-course-metadata batch job, which syncs all data between edx-platform and course-discovery.
+
+Refresh-course-metadata functions differently depending on whether or not the system has Publisher enabled. Specifically, after requesting information from the /courses endpoint in Studio, Discovery will ignore certain fields if Publisher is enabled. These fields are mostly part of the `media` attribute of the API response. They are sent in a variety of ways (absolute urls, paths, sometimes divided by size, etc.)
+
+
+Decision
+--------
+
+The COURSE_CATALOG_INFO_CHANGED will only contain the information necessary to work in a Publisher-enabled environment. In particular, this means it will not contain the `media` field usually present in the Studio /courses API endpoint.
+
+In Discovery, if Publisher is not enabled, the consumer will simply ignore the event and not try to update anything.
+
+Rationale
+---------
+
+The way we update media information in refresh-course-metadata is quite haphazard, with no real standard way of sending over the information or of storing it on the Discovery end. Replicating these structures in the COURSE_CATALOG_INFO_CHANGED signal would make the data definition confusing. Moreover, edx.org has Publisher enabled and would thus ignore these fields anyway.
```

Comment:
Rather than mentioning edx.org, is it accurate to say we believe that Refresh-course-metadata is only used by organizations with Publisher enabled. (Maybe we can circulate this assumption and ADR to double-check?)

Reply:
I actually don't think that's true. Among other things, RCM is more important when there is no publisher since people are more likely to be editing course dates and media in studio.

Reply:
How about something like?:
```suggestion
The way we update media information in refresh-course-metadata is quite haphazard, with no real standard way of sending over the information or of storing it on the Discovery end. Replicating these structures in the COURSE_CATALOG_INFO_CHANGED signal would make the data definition confusing.

Until these fields are needed by a consumer in a non-Publisher environment (if ever), it makes sense to defer trying to find a good solution to this problem.
```